openmc/plotter.py

```diff
@@ -0,0 +1,910 @@
+        None will be returned as the figure and axes have already been
+        generated.
+
+    """
+    import matplotlib.pyplot as plt
+
+    cv.check_type("plot_CE", plot_CE, bool)
+
+    if data_type is None:
+        if isinstance(this, openmc.Nuclide):
+            data_type = 'nuclide'
+        elif isinstance(this, openmc.Element):
+            data_type = 'element'
+        elif isinstance(this, openmc.Material):
+            data_type = 'material'
+        elif isinstance(this, openmc.Macroscopic):
+            data_type = 'macroscopic'
+        elif isinstance(this, str):
+            if this[-1] in string.digits:
+                data_type = 'nuclide'
+            else:
+                data_type = 'element'
+        else:
+            raise TypeError("Invalid type for plotting")
+
+    if plot_CE:
+        # Calculate for the CE cross sections
+        E, data = calculate_cexs(this, data_type, types, temperature, sab_name,
+                                 ce_cross_sections, enrichment)
+        if divisor_types:
+            cv.check_length('divisor types', divisor_types, len(types))
+            Ediv, data_div = calculate_cexs(this, divisor_types, temperature,
+                                            sab_name, ce_cross_sections,
+                                            enrichment)
+
+            # Create a new union grid, interpolate data and data_div on to that
+            # grid, and then do the actual division
+            Enum = E[:]
+            E = np.union1d(Enum, Ediv)
+            data_new = np.zeros((len(types), len(E)))
+
+            for line in range(len(types)):
+                data_new[line, :] = \
+                    np.divide(np.interp(E, Enum, data[line, :]),
+                              np.interp(E, Ediv, data_div[line, :]))
+                if divisor_types[line] != 'unity':
+                    types[line] = types[line] + ' / ' + divisor_types[line]
+            data = data_new
+    else:
+        # Calculate for MG cross sections
+        E, data = calculate_mgxs(this, data_type, types, orders, temperature,
+                                 mg_cross_sections, ce_cross_sections,
+                                 enrichment)
+
+        if divisor_types:
+            cv.check_length('divisor types', divisor_types, len(types))
+            Ediv, data_div = calculate_mgxs(this, data_type, divisor_types,
+                                            divisor_orders, temperature,
+                                            mg_cross_sections,
+                                            ce_cross_sections, enrichment)
+
+            # Perform the division
+            for line in range(len(types)):
+                data[line, :] /= data_div[line, :]
+                if divisor_types[line] != 'unity':
+                    types[line] += ' / ' + divisor_types[line]
+
+    # Generate the plot
+    if axis is None:
+        fig = plt.figure(**kwargs)
+        ax = fig.add_subplot(111)
```

Comment:
Question: Any significance to `111` here?  Or is it a preferred scaling factor or something related to the resulting plotting visual?